### PR TITLE
[MDB Ignore] Port over Zoomba Cyborg sprite

### PR DESCRIPTION
**Pull request by Mooshimi**
## About The Pull Request
Ports cyborg sprites in the "Zoomba" model from Skyrat for:

<details><summary>- Janitor</summary>
<p>

image

</p>
</details>
( Credit for sprites to [Verkister](https://github.com/Verkister) and [Very-Soft](https://github.com/Very-Soft).  ) 

## Why It's Good For The Game

More cyborg icon is good.

(not the janitor sprite)
dreamseeker-17 14 41 458

## Changelog

:cl: Mooshimi, Sprites by Verkister and Very-Soft
add: Ported in Zoomba sprite for janiborgs
/:cl:

### CODE REVIEW

**Comment by thgvr** [comment]
I pray there is a god.

**Comment by Mooshimi** [comment]
I did a good amount of testing to make sure the sprites work alright with the lights!

**Review by MrMelbert — CHANGES_REQUESTED** [review]
I'd be ok with these if you limit it to just janiborg and maybe the service model.

**Comment by Jolly-66** [comment]
Don't forget to attribute your CL for the spriters

**Comment by LemonInTheDark** [comment]
More reskins?

**Comment by Iamgoofball** [comment]
These need to be way, way bigger to click on. They're already an issue at Skyrat due to sprite size. Give them an enlarged hitbox or something.

**Comment by Iamgoofball** [comment]
>Do you still think this with only the janitor sprite included??

yes

File: code/modules/mob/living/silicon/robot/robot_model.dm [hunk 1]
```diff
@@ -746,8 +750,8 @@
 	borg_skins = list(
 		"Bro" = list(SKIN_ICON_STATE = "brobot"),
 		"Butler" = list(SKIN_ICON_STATE = "service_m"),
-		"Kent" = list(SKIN_ICON_STATE = "kent", SKIN_LIGHT_KEY = "medical", SKIN_HAT_OFFSET = 3),
-		"Tophat" = list(SKIN_ICON_STATE = "tophat", SKIN_LIGHT_KEY = NONE, SKIN_HAT_OFFSET = INFINITY),
+		"Kent" = list(SKIN_ICON_STATE = "kent", special_light_key = "medical", SKIN_HAT_OFFSET = 3),
+		"Tophat" = list(SKIN_ICON_STATE = "tophat", special_light_key = NONE, SKIN_HAT_OFFSET = INFINITY),
```
[the remark this answers is not in the document]
Comment (Mooshimi) [reply code/modules/mob/living/silicon/robot/robot_model.dm:754]: The light sprite didn't work with the SKIN_LIGHT_KEY but the var created by Fikou in the refactor special_light_key seen at the top of the file worked fine, I updated these thinking they had the same goal in mind

File: code/modules/mob/living/silicon/robot/robot_model.dm [hunk 2]
```diff
@@ -473,8 +477,8 @@
 	//This is basically just about adding a shake to the borg, effect should look ilke an engine's running
 	var/base_x = robot_owner.base_pixel_x
 	var/base_y = robot_owner.base_pixel_y
-	robot_owner.pixel_x = base_x + rand(-7, 7)
-	robot_owner.pixel_y = base_y + rand(-7, 7)
+	robot_owner.pixel_x = base_x + rand(-13, 7)
+	robot_owner.pixel_y = base_y + rand(-13, 7)
```
[the remark this answers is not in the document]
Comment (Mooshimi) [reply code/modules/mob/living/silicon/robot/robot_model.dm:481]: the image above with the firehat, the cyborg is low on the tile.

**Comment by Mooshimi** [comment]
> These need to be way, way bigger to click on. They're already an issue at Skyrat due to sprite size. Give them an enlarged hitbox or something.

So how in the world would you do this, my boy?

**Comment by Iamgoofball** [comment]
254 alpha pixels around it in the shape of a normal cyborg

File: code/modules/mob/living/silicon/robot/robot_model.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (LemonInTheDark) [reply code/modules/mob/living/silicon/robot/robot_model.dm:754]: It won't actually
`list(variablename = "value")`
is parsed as `list["variablename"] = value` in what the reference calls a "convenient shorthand"
image

**Comment by Dorsisdwarf** [comment]
Gonna be real this looks like a robotics bot, not a cyborg.

**Comment by davethwave** [comment]
Sprite definitely needs to be larger, too many players would choose this just for the smaller size to make getting flashed or hurt harder.

File: code/modules/mob/living/silicon/robot/robot_model.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (Mooshimi) [reply code/modules/mob/living/silicon/robot/robot_model.dm]: Reverted the unrelated borgs i touched but the roomba sprite did not work with lights using SKIN_LIGHT_KEY during my testing so I will keep that as-is unless given further reason

**Review by Twaticus — APPROVED** [review]
i didnt realize this was up for so long, apologies

**Comment by JohnFulpWillard** [comment]
this looks way closer to a bot than a Cyborg to me, I personally would rather see this as some version of the cleanbot instead.

**Comment by Twaticus** [comment]
yeah i agree with john, itd be cute as a reskin for the various bots. 
but as an easily accessible playable mob thats meant to replace a normal human? that sounds like a bad idea. theyre ridiculously small and no alpha pixels to expand the hitbox will remedy that

**Comment by Iamgoofball** [comment]
we literally expanded the hitbox with alpha pixels and players like these skins for cyborgs, why don't we try it before writing it off

**Comment by JohnFulpWillard** [comment]
> we literally expanded the hitbox with alpha pixels and players like these skins for cyborgs, why don't we try it before writing it off

Their hitbox was irrelevant to my point.

You can feel free to re-PR this as a bot/bot-skin, rather than a cyborg. I don't think Borgs, who have an entire toolset including a flash and the ability to pick people up, should look like small bots that can only move around and clean stuff.

**Pull request closed without merging**